filament/backend/src/BackendUtils.cpp

```diff
@@ -16,9 +16,53 @@
 
 #include "private/backend/BackendUtils.h"
 
+#include <string_view>
+
 namespace filament {
 namespace backend {
 
+bool requestsGoogleLineDirectivesExtension(const char* shader, size_t length) noexcept {
+    std::string_view s(shader, length);
+    return s.find("GL_GOOGLE_cpp_style_line_directive") != std::string_view::npos;
+}
+
+void removeGoogleLineDirectives(char* shader, size_t length) noexcept {
+    std::string_view s(shader, length);
+
+    size_t pos = 0;
+    while (true) {
+        pos = s.find("#line", pos);
+        if (pos == std::string_view::npos) {
+            break;
```

Comment:
Doesn't this mean you will only fix the first #line directive?

Reply:
No, this breaks when there are no more #line directives in the string.

Reply:
Ah right of course. I keep thinking you're doing processing line by line.